code/datums/components/orbit_poll.dm

```diff
@@ -0,0 +1,114 @@
+/**
+ * A replacement for the standard poll_ghost_candidate.
+ * Use this to subtly ask players to join - it picks from orbiters.
+ * Please use named arguments for this.
+ *
+ * @params ignore_key - Required so it doesn't spam
+ * @params job_bans - You can insert a list or single items here.
+ * @params cb - Invokes this proc and appends the poll winner as the last argument, mob/dead/observer/ghost
+ * @params title - Optional. Useful if the parent doesn't have the name you want to use.
+ *
+ * @usage
+ * ```
+ * var/datum/callback/cb = CALLBACK(src, PROC_REF(do_stuff), arg1, arg2)
+ * AddComponent(/datum/component/orbit_poll, \
+ *   ignore_key = POLL_IGNORE_EXAMPLE, \
+ *   job_bans = ROLE_EXAMPLE or list(ROLE_EXAMPLE, ROLE_EXAMPLE2), \
+ *   title = "Use this if you want something other than the parent name", \
+ *   cb = cb, \
+ * )
+ */
+/datum/component/orbit_poll
+	/// Prevent players with this ban from being selected
+	var/list/ineligible_roles = list()
+	/// Title of the role to announce after it's done
+	var/role_name
+	/// Proc to invoke whenever the poll is complete
+	var/datum/callback/to_call
+
+/datum/component/orbit_poll/Initialize( \
+	ignore_key, \
+	list/job_bans, \
+	datum/callback/cb, \
```

Comment:
It's our usual standard to have the init arguments match the var name they are stored in, so in this case would do
`src.to_call = to_call`

Perhaps the same with `job_bans` as it doesn't seem like the list is mutated